sqllogictest/src/parser.rs

```diff
@@ -765,14 +792,17 @@ fn parse_inner<T: ColumnType>(loc: &Location, script: &str) -> Result<Vec<Record
                     connection: std::mem::take(&mut connection),
                     sql,
                     expected,
+                    retry,
                 });
             }
             ["query", res @ ..] => {
-                let mut expected = match res {
-                    ["error", tokens @ ..] => {
-                        let error = ExpectedError::parse_inline_tokens(tokens)
+                let (mut expected, res) = match res {
+                    ["error", err_tokens @ ..] => {
+                        // NOTE: `query error` can't be used with `retry` now because all the tokens
+                        // after `error` are treated as error message.
```

Comment:
Not sure whether we need it in RisingWave. FYI @xxchan

related #102

Reply:
Retry with expected error doesn't seem to be very useful